Mathlib/Data/Fintype/Pi.lean

```diff
@@ -109,3 +110,18 @@ theorem Finset.univ_pi_univ {α : Type*} {β : α → Type*} [DecidableEq α] [F
     (Finset.univ.pi fun a : α => (Finset.univ : Finset (β a))) = Finset.univ := by
   ext; simp
 #align finset.univ_pi_univ Finset.univ_pi_univ
+
+@[simp]
+lemma Fin.cons_mem_piFinset_iff {α} {n : ℕ} (p : (Fin n → α)) (x : α) (S : Finset α) :
+    Fin.cons x p ∈ Fintype.piFinset (fun _ => S) ↔ x ∈ S ∧ p ∈ Fintype.piFinset (fun _ => S) := by
```

Comment:
Can you generalize `fun _ => S` here to `S : Fin n → Finset α`? I think that gives
```suggestion
    Fin.cons x p ∈ Fintype.piFinset (fun _ => S) ↔ x ∈ S 0 ∧ p ∈ Fintype.piFinset (fun i => S i.succ) := by
```

Reply:
Ok will do

Reply:
Actually I thought it looked asymmetrical to have cons on the LHS but not the right, so I split into two versions, one with cons and one without.